syntax = "proto3";

package track.search;

message SearchTracksRequest {
  string title = 1;
  string artist_name = 2;
  repeated string genre_name = 3;
  int32 min_duration_ms = 4;
  int32 max_duration_ms = 5;
  bool explicit = 6;
  string release_date_from = 7; // ISO-строка yyyy-MM-dd
  string release_date_to = 8;   // ISO-строка yyyy-MM-dd
  int32 page = 9;
  int32 page_size = 10;
}

message TrackItem {
  int64 track_id = 1;
  string title = 2;
  int32 duration_ms = 3;
  repeated string artists = 4;
  repeated string genres = 5;
  bool explicit = 6;
  string release_date = 7;
}

message SearchTracksResponse {
  repeated TrackItem tracks = 1;
  int32 total = 2;
  int32 page = 3;
  int32 page_size = 4;
  bool success = 5;
}



service TrackSearchService {
  rpc Search(SearchTracksRequest) returns (SearchTracksResponse);
}
